src/main/java/carpet/mixins/ShulkerBoxBlockEntity_renderFixMixin.java

```diff
@@ -0,0 +1,30 @@
+package carpet.mixins;
```

Comment:
why do you need that? Can you explain what's wrong with it?

Reply:
here is some shulkers block s render code in  net.minecraft.client.renderer.blockentity.ShulkerBoxRenderer
it get its block state from its level.
```
	public void render(ShulkerBoxBlockEntity shulkerBoxBlockEntity, float f, PoseStack poseStack, MultiBufferSource multiBufferSource, int i, int j) {
		Direction direction = Direction.UP;
		if (shulkerBoxBlockEntity.hasLevel()) {
			BlockState blockState = shulkerBoxBlockEntity.getLevel().getBlockState(shulkerBoxBlockEntity.getBlockPos());
			if (blockState.getBlock() instanceof ShulkerBoxBlock) {
				direction = blockState.getValue(ShulkerBoxBlock.FACING);
			}
		}
```

but other block entities get it more directly:
here is beds
```
	public void render(BedBlockEntity bedBlockEntity, float f, PoseStack poseStack, MultiBufferSource multiBufferSource, int i, int j) {
		Material material = Sheets.BED_TEXTURES[bedBlockEntity.getColor().getId()];
		Level level = bedBlockEntity.getLevel();
		if (level != null) {
			BlockState blockState = bedBlockEntity.getBlockState();
```
so i have to use a mixin to fix that. as there is not a real shulker box in the world.

Reply:
i read others block entities code. only shulker box write it that way.

Reply:
(if only you can change shulker box`s code, then this file is no longer needed.)

Reply:
problem is you are changing the vanilla code for some use of some mod. If you change this, does it mess up rendering of shulker boxes in some other places? what about in the inventory? All in all, carpet should strive to preserve vanilla implementation at all cost, even if it ends up in ugly code, or causing things to not be perfect.

Reply:
i tried. and i did not find any problem in rendering at all.

and i think is the one who write that code forget that he could get it directly. XD

Reply:
fixed this in another way.
this time no vanilla code changed.